test/AspNet.Security.OAuth.Providers.Tests/Huawei/HuaweiTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+namespace AspNet.Security.OAuth.Huawei;
+
+public class HuaweiTests : OAuthTests<HuaweiAuthenticationOptions>
+{
+    public HuaweiTests(ITestOutputHelper outputHelper)
+    {
+        OutputHelper = outputHelper;
+    }
+
+    public override string DefaultScheme => HuaweiAuthenticationDefaults.AuthenticationScheme;
+
+    protected internal override void RegisterAuthentication(AuthenticationBuilder builder)
+    {
+        builder.AddHuawei(options =>
+        {
+            options.GetNickName = true;
+
+            options.Scope.Add("profile");
+            options.Scope.Add("email");
+
+            ConfigureDefaults(builder, options);
+        });
+    }
+
+    [Theory]
+    [InlineData(ClaimTypes.NameIdentifier, "test-name-identifier")]
+    [InlineData(ClaimTypes.Name, "test-display-name")]
+    [InlineData(HuaweiAuthenticationConstants.Claims.Avatar, "test-head-picture-url.jpg")]
+    [InlineData(ClaimTypes.Email, "test-email@test")]
```

Comment:
Could you extend the default response to include the nickname, add a default claim action to map it, and then assert that it's here in the test please?

Reply:
![image](https://user-images.githubusercontent.com/1539280/199233247-1cf3d20e-5ff1-42e6-bb75-0bf7c92ba5dd.png)

![image](https://user-images.githubusercontent.com/1539280/199234072-fe32bcba-4b48-41d6-b2f8-cf2002582970.png)

'Indicates whether the nickname is returned preferentially.' Just like: displayName = getNickName == true ? "Vicente" : "y****x****@***.com"; Only one field named 'displayName'.  :)